lib/charms/bind/v0/dns_record.py

```diff
@@ -284,21 +278,20 @@ def set_password_id(self, model: ops.Model, relation: ops.Relation) -> None:
             model: the Juju model
             relation: relation to grant access to the secrets to.
         """
+        # password is always defined since pydantic guarantees it
+        password = cast(SecretStr, self.password)
```

Comment:
While we're into renaming some fields, and since we're now using "ServiceAccount" shouldn't we rename that one to "auth_key" ?

Reply:
Will send just an auth key